### PR TITLE
Cache the default malloc zone

### PR DESCRIPTION
**Pull request by Catfish-Man**
I measured a 7% win on FilterEvenUsingReduceInto with fast dealloc enabled (which it won't be in CI)

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
Looks like with fast dealloc disabled the win is below the 7% threshold to be reported, so we just get noise.

Disturbing to see (?) on a 36% change though.

File: stdlib/public/runtime/Heap.cpp [hunk 1]
```diff
@@ -57,6 +61,13 @@ using namespace swift;
 static_assert(_swift_MinAllocationAlignment > MALLOC_ALIGN_MASK,
               "Swift's default alignment must exceed platform malloc mask.");
 
+#if defined(__APPLE__)
+static inline malloc_zone_t *DEFAULT_ZONE() {
+  static malloc_zone_t *z = SWIFT_LAZY_CONSTANT(malloc_default_zone());
```
Comment (gribozavr) [inline stdlib/public/runtime/Heap.cpp]: The `static` and `SWIFT_LAZY_CONSTANT` create two levels of lazy initialization; we only need one of them IIUC.

Comment (mikeash) [reply stdlib/public/runtime/Heap.cpp]: Good catch! This shouldn't have the `static`.

**Pull request merged**